Emphasize permitted version format in packages for publication

OlenaYefymenko opened the pull request:
The patch adds mention in README that local version identifiers in packages don’t allow publishing to PyPI/TestPyPI. Also, it suggests alternative local servers for using local version identifiers.

Resolves #322

OlenaYefymenko commented:
@webknjaz, could you please review the updated PR and provide your feedback?

webknjaz commented on README.rst:
```diff
@@ -292,6 +292,16 @@ the tag.
+
+    According to `PEP 440 <https://peps.python.org/pep-0440/#local-version-identifiers>`_,
+    if a version includes a local component, the package cannot be published to public
+    package indexes like PyPI or TestPyPI. Usually,non-permitted version formats may
+    occur due to auto-publishing or by mistake.
```
```suggestion
    package indexes like PyPI or TestPyPI. The disallowed version segments may
    be seen in auto-publishing workflows or when a configuration mistake is made.
```

webknjaz commented on README.rst:
```diff
@@ -292,6 +292,16 @@ the tag.
+    occur due to auto-publishing or by mistake.
+
+    However, if you need to use local version identifiers in your packages, you
+    can use a local server such as ``devpi`` or other alternatives that allow local
```
```suggestion
    can use a self-hosted package index such as ``devpi`` or other alternatives that allow local
```

webknjaz commented on README.rst:
```diff
@@ -292,6 +292,16 @@ the tag.
+
+    However, if you need to use local version identifiers in your packages, you
+    can use a local server such as ``devpi`` or other alternatives that allow local
+    versions. Local version identifiers must comply with PEP 440.
```
I believe, it's possible to reuse the link from above like this:
```suggestion
    versions. Local version identifiers must comply with `PEP 440`_.
```

webknjaz commented on README.rst:
```diff
@@ -292,6 +292,16 @@ the tag.
+    package indexes like PyPI or TestPyPI. The disallowed version segments may
+    be seen in auto-publishing workflows or when a configuration mistake is made.
+
+    However, if you need to use local version identifiers in your packages, you
```
I think that we shouldn't suggest people to use devpi just for the local version. This motivation is probably wrong. Instead, mention that some indexes, like devpi, allow those.

OlenaYefymenko committed:
Emphasize permitted version format in packages for publication

The patch adds mention in README that local version identifiers in packages
don't allow publishing to PyPI/TestPyPI. Also, it suggests alternative local
servers for using local version identifiers.

Resolves #322

Co-authored-by: Sviatoslav Sydorenko <webknjaz+github/profile@redhat.com>

OlenaYefymenko committed:
Edit typo in README.rst

Co-authored-by: Sviatoslav Sydorenko <wk.cvs.github@sydorenko.org.ua>
